docs/DeveloperGuide.adoc

```diff
@@ -317,7 +316,238 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+Step 1. The user launches the application for the first time. The `StorageController#retrieveData()` will retrieve all datasets saved locally.
+
+Step 2. The user executes `class list` command to list all classrooms to Trajectory. The `class list` command calls the `ClassListCommand#execute()`. The `ClassroomManager` will be instantiated and read the classroom list from the storage and converts the data from XmlAdaptedClassroom to the Classroom data type.
+
+Step 3. The classroom list with the corresponding classroom information will be appended to the with the support of the `StringBuilder` and displayed as a message successfully.
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+image::classaddlist.PNG[width="800"]
+
+//zcend
+
+// tag::datatograph[]
+=== Gradebook Graph feature
+==== Current Implementation
+
+The gradebook graph mechanism is an enhancement that will be released in the later versions, facilitated by 'Trajectory'.
+It is stored internally in GradebookManager.
+
+Additionally, it implements the following operations:
+
+* `gradebookManager#graphModuleSummary()` -- Converts data of all student grades from Array List to graph form
+* `gradebookManager#graphStudentProgress()` -- Converts student data to present progress on module.
+
+These operations are exposed in the `GradebookManager` as `GradebookManager#graphModuleSummary()`, `GradebookManager#graphStudentProgress()` respectively.
+
+Given below is an example usage scenario and how the gradebook data-to-graph mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The StorageController which interacts with #xmlAdaptedGradebook to retrieve data from Array List using #retrieveData.
+
+Step 2. The user executes `gradebook find m/cs2113 i/Finals` command to find the relevant gradebook component. The `find` command calls `GradebookManager#findGradebookComponent()`, which finds and filters the Array List to the relevant search.
+
+Step 3. The user executes `gradebook graph student`. GradebookManager#graphStudentProgress will convert the Array List to graph form and display to the user.
+
+[NOTE]
+If a command fails its execution, it will not call `Gradebook#GradebookManager()`, so Trajectory state will not be saved into the `GradebookManager`.
+
+Step 4. The user now decides to export graph according to the progress of a student, and that action is done by executing the `gradebook graph student` command. This command will call `GradebookManager#graphStudentProgress()`, which then displays the graph of the students progress.
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+image::dataToGraphActivityDiagram.png[width="650"]
+
+==== Design Considerations
+
+===== Aspect: How graph-to-data executes
+
+* **Alternative 1 (current choice):** Individual command knows how to export accordingly.
```

Comment:
You may consider rewording this. 

A command 'does not know' how to export. Executing the command results in (Something) etc.